storage-proofs/src/hasher/pedersen.rs

```diff
@@ -48,12 +57,48 @@ impl Hasher for PedersenHasher {
     }
 }
 
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
-pub struct PedersenFunction(Fr);
+#[derive(Clone, PartialEq)]
+pub struct PedersenFunction {
+    // The Pedersen Hash segment index of the next unhashed segment.
+    next_segment: u32,
+    // Stores any unhashed bits between calls to `self.update()`. `unhashed_bits`'s bit-length never
+    // exceeds `N_SEGMENT_BITS` (i.e. never contains more than one segment's worth of unhashed
+    // data).
+    unhashed_bits: BitVec<LittleEndian, u8>,
```

Comment:
Can we drop this, if we enforce that always 32 bytes will be "updated"?

Reply:
It could be dropped if updates always occurred in multiples of 189 bits. As currently written, `unhashed_bits` will never be larger than 188 bits long.